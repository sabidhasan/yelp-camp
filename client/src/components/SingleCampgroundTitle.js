import React from 'react'
import PropTypes from 'prop-types'

import withStickyBar from './withStickyBar'

const SingleCampgroundTitle = (props, context) => {
  return (
    <h1 className={`singleCampground__title ${context.stickyClass}`}>
      {props.name}
      <span className='singleCampground__region'>
        {props.region} Region
      </span>
    </h1>
  )
}

SingleCampgroundTitle.contextTypes = {
  stickyClass: PropTypes.string,
};

export default withStickyBar(SingleCampgroundTitle)
