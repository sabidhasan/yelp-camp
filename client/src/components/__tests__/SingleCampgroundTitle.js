import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import SingleCampgroundTitle from '../SingleCampgroundTitle'

configure({ adapter: new Adapter() });

describe('<SingleCampgroundTitle />', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      name: 'Title',
      region: 'Test'
    }
    component = mount(<SingleCampgroundTitle { ...props } />, {context: {stickyClass: 'sampleClass'}})
  })

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });

  it('Region text should be correctly displayed', () => {
    expect(component.find('.SingleCampgroundTitle').text()).toBe('Title');
  });

  it('Region text should be correctly displayed', () => {
    expect(component.find('.SingleCampgroundTitle__region').text()).toBe('Test Region');
  });

  it('SingleCampgroundTitle component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
})
