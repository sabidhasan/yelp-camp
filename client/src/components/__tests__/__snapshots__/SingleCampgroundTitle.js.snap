// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleCampgroundTitle /> SingleCampgroundTitle component snapshot matches what is expected 1`] = `
<withStickyBar
  name="Title"
  region="Test"
>
  <SingleCampgroundTitle
    name="Title"
    region="Test"
  >
    <h1
      aria-label="Campground name"
      className="SingleCampgroundTitle "
    >
      Title
    </h1>
    <h2
      aria-label="Campground region"
      className="SingleCampgroundTitle__region"
    >
      Test
       Region
    </h2>
  </SingleCampgroundTitle>
</withStickyBar>
`;
